Wavernn tuning

RuntimeRacer opened the pull request:
This PR adds a variety of changes in order to improve Vocoder Inference speed & usability in Real-Time voice synthesis applications. Most improvements have been adapted from geneing's awesome WaveRNN Repo: https://github.com/geneing/WaveRNN-Pytorch

General Overview of changes made:

- Generalization of Vocoder Model components & the way synthesizers are being initialized when needed.
- Added WaveRNN Implementation from geneing's Repo + an adapted version by myself which aims at maximizing quality with improved inference speed.
- Added libwavernn from geneing's repo & Improved CMakeList for simpler sourcing & compilation process.
- Added a build script for building libwavernn
- Added my own adapted version of libwavernn, tailored to work with my adapted pytorch WaveRNN Model & allow for Multiprocessing
- Integrated unstructured model Pruning process from geneing's repo for WaveRNN Pytorch model training process.
- Added a simple bash script for installing dependencies and building libwavernn.
- Added helper scripts for testing libwavernn and performing vocoder model binary conversion.
- Added a Wrapper class for handling Libwavernn and performing multi-core inference on pruned vocoder models.
- Added a toggle in Voice cloning Toolbox to switch to binary / libwavernn mode and use the C++ processing instead.

Supplementary notes / Todo Remarks:

- Fatchord WaveRNN Model won't work out of the box with pruning, since the binary conversion & C++ code excepts input to be divisible by 8. However, even when making necessary adaptions i ran into weird errors when trying to do inference with a 512-channel RNN network in libwavernn as well. For that reason I changed my implementation to do 256-channel RNN-Chaining instead, which didn't have these issues and achieving similar output quality.
- Currently there is no extended documentation included on how to use the libwavernn functionalities. I'll try to provide these soon.
- A lot of this code has not yet received a major cleanup. I will try to do this alongside the documentation when I'll do that.

RuntimeRacer committed:
Initially added Forward Tacotron and FastPitch Models + applied some simplifications

RuntimeRacer committed:
Applied modifications to Forward tacotron as in Legacy version (https://github.com/as-ideas/ForwardTacotron/commit/4daa7014fe1f7ff58234f571c1f2d8f346dbbeb5) to allow for Multispeaker training.

RuntimeRacer committed:
Small refactoring. But I think I will switch this all to YAML like in ForwardTacotron repo...

RuntimeRacer committed:
Merge branch 'master' into forward-multispeaker

# Conflicts:
#	vocoder_preprocess.py

RuntimeRacer committed:
extracted all hparams (except encoder) into a global hparam module and make everything use this global config. This reduces flexibility, however this level of flexibility is rarely needed.

RuntimeRacer committed:
Preparations for Forward Tacotron init & training

RuntimeRacer committed:
Preparations for Forward Tacotron training

RuntimeRacer committed:
WIP

RuntimeRacer committed:
small improvements to training schedules

RuntimeRacer committed:
Merge branch 'master' into forward-multispeaker

# Conflicts:
#	synthesizer/hparams.py
#	vocoder/hparams.py

RuntimeRacer committed:
Merge branch 'master' into forward-multispeaker

RuntimeRacer committed:
first approach on generating alignment features for foward tacotron - not ready to be tested yet.

RuntimeRacer committed:
Working alignment preprocessing - very slow.

RuntimeRacer committed:
Some improvements towards parallelization, but still slow. Needs profiling.

RuntimeRacer committed:
Basic training for tacotron & forward tacotron ported over. Untested. Sample generation & progress stream need fixing.

RuntimeRacer committed:
Plot generation for Forward tacotron - untested

RuntimeRacer committed:
Fixed a bunch of issues to make training work. Encountered some bug with Preprocessing which is still unfixed.

RuntimeRacer committed:
simplified preprocessing method to avoid memory pointer issues which seem to corrupt alignment output. Still slow but should be a bit more reliable.

RuntimeRacer committed:
Heureka! First fully functioning prototype of Forward Tacotron Training + Sample Generation.

RuntimeRacer committed:
Fix length of generated audio samples

RuntimeRacer committed:
Fixing issues slowing down quick processing of alignment feature creation (#5)

Co-authored-by: RuntimeRacer <runtimeracer@gmail.com>

RuntimeRacer committed:
re-added missing datasets for synthesizer

RuntimeRacer committed:
small fix

RuntimeRacer committed:
Fixing some processing output issues

RuntimeRacer committed:
fixing warnings

RuntimeRacer committed:
another attempt to fix progress bars

RuntimeRacer committed:
revert smoothing change

RuntimeRacer committed:
small fixes and adaptions required for training, to allow dynamic UI change based on model type in toolbox later.

RuntimeRacer committed:
fix processing warning

RuntimeRacer committed:
fix processing warning

RuntimeRacer committed:
accelerator quicktry

RuntimeRacer committed:
accelerator quicktry

RuntimeRacer committed:
accelerator quicktry

RuntimeRacer committed:
accelerator quicktry

RuntimeRacer committed:
use torch pool

RuntimeRacer committed:
detail

RuntimeRacer committed:
remove no longer needed utterance IDs as dataset output

RuntimeRacer committed:
slowly increasing batch size

RuntimeRacer committed:
unwrap before generating plots

RuntimeRacer committed:
produce multiple variations when sampling based on pitch and energy

RuntimeRacer committed:
typo

RuntimeRacer committed:
typo & better modifier

RuntimeRacer committed:
stronger weighting for energy modifier

RuntimeRacer committed:
Added forward taco tuning UI elements and modified layout a bit.

RuntimeRacer committed:
First approach on using forward tacotron via toolbox

RuntimeRacer committed:
small fixes

RuntimeRacer committed:
fix LR

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
add skip existing param to embedding creation

RuntimeRacer committed:
Prepared Vocoder preprocessing using forward tacotron

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
voc params

RuntimeRacer committed:
details

RuntimeRacer committed:
Added optimized WaveRNN model. Generate function not ready yet

RuntimeRacer committed:
fixed LR for latest update.

RuntimeRacer committed:
first attempt on making geneing's version of WaveRNN trainable

RuntimeRacer committed:
hotfix for testing

RuntimeRacer committed:
vocoder hparams fix

RuntimeRacer committed:
vocoder hparams fix

RuntimeRacer committed:
vocoder hparams fix

RuntimeRacer committed:
vocoder hparams fix

RuntimeRacer committed:
vocoder hparams fix

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
small fixes for training

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
remove layer update. Maybe it is not even needed.

RuntimeRacer committed:
no performance boost, revert

RuntimeRacer committed:
pruning in all threads; only during training step itself and not while saving / eval

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
small optimization on reaching prune start faster

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
add griffin-lim output for comparison (to check if the audio / mel itself might be broken)

RuntimeRacer committed:
add griffin-lim output for comparison (to check if the audio / mel itself might be broken)

RuntimeRacer committed:
add griffin-lim output for comparison (to check if the audio / mel itself might be broken)

RuntimeRacer committed:
add griffin-lim output for comparison (to check if the audio / mel itself might be broken)

RuntimeRacer committed:
add griffin-lim output for comparison (to check if the audio / mel itself might be broken)

RuntimeRacer committed:
add griffin-lim output for comparison (to check if the audio / mel itself might be broken)

RuntimeRacer committed:
add griffin-lim output for comparison (to check if the audio / mel itself might be broken)

RuntimeRacer committed:
add griffin-lim output for comparison (to check if the audio / mel itself might be broken)

RuntimeRacer committed:
not sure if this has any effect, but for testing...

RuntimeRacer committed:
store model type in save

RuntimeRacer committed:
modified anomaly detection and extended it with a recovery function. Needs to be tested.

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
trying something

RuntimeRacer committed:
trying something

RuntimeRacer committed:
detail

RuntimeRacer committed:
add libwavernn and tweak hparams for finetuning of waveRNN

RuntimeRacer committed:
trying to implement blacklisting with global variable

RuntimeRacer committed:
learning rate fixes

RuntimeRacer committed:
reset learning rates for better quality dataset

RuntimeRacer committed:
detail

RuntimeRacer committed:
Fix anomaly detection related instructions being performed all the time + too early benchmark evaluation for log stream

RuntimeRacer committed:
disable everything that is not needed initially for best performance

RuntimeRacer committed:
details + lr adjustments

RuntimeRacer committed:
detail adjustments for MOL training

RuntimeRacer committed:
detail adjustments for MOL training

RuntimeRacer committed:
detail

RuntimeRacer committed:
- Since batch size is per GPU, do not divide by amount of devices
- Slight changes to learning rate and upsampling values for testing how this affects development of the loss

RuntimeRacer committed:
Reducing Batch size for quicker step progression and increasing loss on late epochs for quicker loss progression

RuntimeRacer committed:
Added binary conversion script

RuntimeRacer committed:
detail

RuntimeRacer committed:
Couple of Adaptions to build the C++ Module. Unfortunately there are still some issues with it.

RuntimeRacer committed:
fix gitignore

RuntimeRacer committed:
Adaptions for Vocoder inference methods to be capable of handling the geneing model

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
Structured Pruning experiments

RuntimeRacer committed:
added custom WaveRNN variant

RuntimeRacer committed:
details

RuntimeRacer committed:
details

RuntimeRacer committed:
Trying once more with proper model sizing

RuntimeRacer committed:
Trying once more with proper model sizing

RuntimeRacer committed:
disable testset for quick test

RuntimeRacer committed:
Revert "disable testset for quick test"

This reverts commit 5ff8e80d4566a44f9e577690a3653915c3a8266c.

RuntimeRacer committed:
disable softmax

RuntimeRacer committed:
experiment using 4xRRN at 256 Dims

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
added additional filter layer

RuntimeRacer committed:
Revert "added additional filter layer"

This reverts commit 7daa620735bd6e7873fb76f7196075d96e3474e7.

RuntimeRacer committed:
maybe order is important

RuntimeRacer committed:
quick prune test

RuntimeRacer committed:
revert a bunch of things

RuntimeRacer committed:
details

RuntimeRacer committed:
details

RuntimeRacer committed:
started working on own C++ variant

RuntimeRacer committed:
detail fixes to allow for compilation

RuntimeRacer committed:
detail fixes to allow for compilation and first adaption for RR variant

RuntimeRacer committed:
Small fixes

RuntimeRacer committed:
Small fixes

RuntimeRacer committed:
fixed flipped method args

RuntimeRacer committed:
Fixes for reading Mel from NPY in C++ and adaptions to make RuntimeRacer WaveRNN work.

RuntimeRacer committed:
Fixes for reading Mel from NPY in C++ and adaptions to make RuntimeRacer WaveRNN work.

RuntimeRacer committed:
Fixes for reading Mel from NPY in C++ and adaptions to make RuntimeRacer WaveRNN work.

RuntimeRacer committed:
Figured out there is only one last issue remaining in MuLaw Inversion

RuntimeRacer committed:
Figured out proper way of applying MuLaw Decode

RuntimeRacer committed:
detail

RuntimeRacer committed:
Merge branch 'wavernn-tuning' of https://github.com/RuntimeRacer/Real-Time-Voice-Cloning into wavernn-tuning

 Conflicts:
	vocoder/libwavernn/runtimeracer_version/src/wavernn.h

RuntimeRacer committed:
some experiments to get performance closer to fatchord version

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
rr v6

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
detail

RuntimeRacer committed:
fixes for cpp impl + hparam tuning

RuntimeRacer committed:
rollback to v4 with minimal changes

RuntimeRacer committed:
Adaptions for v4a

RuntimeRacer committed:
A lot of cleanup

RuntimeRacer committed:
- First implementation for LibWaveRNN Inference wrapper
- Smaller cleanups and fixes

RuntimeRacer committed:
Working libwavernn high speed multicore inference

RuntimeRacer committed:
Added libwavernn vocoding to toolbox. Also Added seed for eigen in C++ code; however not sure if this actually does anything...

RuntimeRacer committed:
detai

RuntimeRacer committed:
Merge branch 'master' into wavernn-tuning

# Conflicts:
#	config/hparams.py
#	synthesizer/synthesize.py
#	toolbox/toolbox.py
#	toolbox/ui.py
#	vocoder/batch.py
#	vocoder/gen_wavernn.py
#	vocoder/inference.py
#	vocoder/train.py
#	vocoder/visualizations.py
#	vocoder/vocoder_dataset.py

RuntimeRacer committed:
cleanup

trholding commented:
Hi 

I had an issue when running the demo_cli on and also building libwavernn on Linux x86_64

```
import WaveRNNVocoder
ModuleNotFoundError: No module named 'WaveRNNVocoder'
```

Changing the build.sh like this got the build started, still building:

```

# Specific Python version in conda env, will fail if no specific version in your env:
if [[ ! -z $CONDA_PREFIX ]]
then
  cmake ../$BUILD_VARIANT/src -DPYTHON_EXECUTABLE=$CONDA_PREFIX/bin/python
else
#  cmake ../$BUILD_VARIANT/src -DPYTHON_EXECUTABLE=$CONDA_PYTHON_EXE
# The is the change:
cmake ../$BUILD_VARIANT/src -DPYTHON_INCLUDE_DIR=$(python -c "from distutils.sysconfig import get_python_inc; print(get_python_inc())") -DPYTHON_LIBRARY=$(python -c "import distutils.sysconfig as sysconfig; print(sysconfig.get_config_var('LIBDIR'))")
fi
```

trholding commented:
Didn't work out, here is the error log in case you feel like debugging:

```
./build.sh
WARNING: Skipping WaveRNNVocoder as it is not installed.
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python3: /usr/bin/python3.10 (found version "3.10.7") found components: Interpreter Development Development.Module Development.Embed 
-- Found PythonInterp: /usr/bin/python (found version "3.10.7") 
-- Found PythonLibs: /usr/lib
-- Performing Test HAS_CPP14_FLAG
-- Performing Test HAS_CPP14_FLAG - Success
-- pybind11 v2.2.3
-- Found ZLIB: /usr/lib/libz.so (found version "1.2.12") 
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11 - Success
-- Performing Test COMPILER_SUPPORT_std=cpp03
-- Performing Test COMPILER_SUPPORT_std=cpp03 - Success
-- Performing Test standard_math_library_linked_to_automatically
-- Performing Test standard_math_library_linked_to_automatically - Success
-- Standard libraries to link to explicitly: none
-- Performing Test COMPILER_SUPPORT_WERROR
-- Performing Test COMPILER_SUPPORT_WERROR - Success
-- Performing Test COMPILER_SUPPORT_pedantic
-- Performing Test COMPILER_SUPPORT_pedantic - Success
-- Performing Test COMPILER_SUPPORT_Wall
-- Performing Test COMPILER_SUPPORT_Wall - Success
-- Performing Test COMPILER_SUPPORT_Wextra
-- Performing Test COMPILER_SUPPORT_Wextra - Success
-- Performing Test COMPILER_SUPPORT_Wundef
-- Performing Test COMPILER_SUPPORT_Wundef - Success
-- Performing Test COMPILER_SUPPORT_Wcastalign
-- Performing Test COMPILER_SUPPORT_Wcastalign - Success
-- Performing Test COMPILER_SUPPORT_Wcharsubscripts
-- Performing Test COMPILER_SUPPORT_Wcharsubscripts - Success
-- Performing Test COMPILER_SUPPORT_Wnonvirtualdtor
-- Performing Test COMPILER_SUPPORT_Wnonvirtualdtor - Success
-- Performing Test COMPILER_SUPPORT_Wunusedlocaltypedefs
-- Performing Test COMPILER_SUPPORT_Wunusedlocaltypedefs - Success
-- Performing Test COMPILER_SUPPORT_Wpointerarith
-- Performing Test COMPILER_SUPPORT_Wpointerarith - Success
-- Performing Test COMPILER_SUPPORT_Wwritestrings
-- Performing Test COMPILER_SUPPORT_Wwritestrings - Success
-- Performing Test COMPILER_SUPPORT_Wformatsecurity
-- Performing Test COMPILER_SUPPORT_Wformatsecurity - Success
-- Performing Test COMPILER_SUPPORT_Wshorten64to32
-- Performing Test COMPILER_SUPPORT_Wshorten64to32 - Failed
-- Performing Test COMPILER_SUPPORT_Wlogicalop
-- Performing Test COMPILER_SUPPORT_Wlogicalop - Success
-- Performing Test COMPILER_SUPPORT_Wenumconversion
-- Performing Test COMPILER_SUPPORT_Wenumconversion - Success
-- Performing Test COMPILER_SUPPORT_Wcpp11extensions
-- Performing Test COMPILER_SUPPORT_Wcpp11extensions - Success
-- Performing Test COMPILER_SUPPORT_Wdoublepromotion
-- Performing Test COMPILER_SUPPORT_Wdoublepromotion - Success
-- Performing Test COMPILER_SUPPORT_Wshadow
-- Performing Test COMPILER_SUPPORT_Wshadow - Success
-- Performing Test COMPILER_SUPPORT_Wnopsabi
-- Performing Test COMPILER_SUPPORT_Wnopsabi - Success
-- Performing Test COMPILER_SUPPORT_Wnovariadicmacros
-- Performing Test COMPILER_SUPPORT_Wnovariadicmacros - Success
-- Performing Test COMPILER_SUPPORT_Wnolonglong
-- Performing Test COMPILER_SUPPORT_Wnolonglong - Success
-- Performing Test COMPILER_SUPPORT_fnochecknew
-- Performing Test COMPILER_SUPPORT_fnochecknew - Success
-- Performing Test COMPILER_SUPPORT_fnocommon
-- Performing Test COMPILER_SUPPORT_fnocommon - Success
-- Performing Test COMPILER_SUPPORT_fstrictaliasing
-- Performing Test COMPILER_SUPPORT_fstrictaliasing - Success
-- Performing Test COMPILER_SUPPORT_wd981
-- Performing Test COMPILER_SUPPORT_wd981 - Failed
-- Performing Test COMPILER_SUPPORT_wd2304
-- Performing Test COMPILER_SUPPORT_wd2304 - Failed
-- Performing Test COMPILER_SUPPORT_STRICTANSI
-- Performing Test COMPILER_SUPPORT_STRICTANSI - Failed
-- Performing Test COMPILER_SUPPORT_Qunusedarguments
-- Performing Test COMPILER_SUPPORT_Qunusedarguments - Failed
-- Performing Test COMPILER_SUPPORT_ansi
-- Performing Test COMPILER_SUPPORT_ansi - Success
-- Performing Test COMPILER_SUPPORT_OPENMP
-- Performing Test COMPILER_SUPPORT_OPENMP - Success
-- Found unsuitable Qt version "5.15.6" from /usr/bin/qmake
-- Looking for a Fortran compiler
-- Looking for a Fortran compiler - /usr/bin/f95
-- The Fortran compiler identification is GNU 12.2.0
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /usr/bin/f95 - skipped
-- Found CHOLMOD: /usr/include  
-- Found UMFPACK: /usr/include  
-- Found KLU: /usr/include  
-- Could NOT find SuperLU (missing: SUPERLU_INCLUDES SUPERLU_LIBRARIES SUPERLU_VERSION_OK) (Required is at least version "4.0")
-- Checking for one of the modules 'hwloc'
-- Performing Test HAVE_HWLOC_PARENT_MEMBER
-- Performing Test HAVE_HWLOC_PARENT_MEMBER - Success
-- Performing Test HAVE_HWLOC_CACHE_ATTR
-- Performing Test HAVE_HWLOC_CACHE_ATTR - Success
-- Performing Test HAVE_HWLOC_OBJ_PU
-- Performing Test HAVE_HWLOC_OBJ_PU - Success
-- Looking for hwloc_bitmap_free in hwloc
-- Looking for hwloc_bitmap_free in hwloc - found
-- Looking for Fortran sgemm
-- Looking for Fortran sgemm - found
-- Performing Test SCOTCH_Num_4
-- Performing Test SCOTCH_Num_4 - Failed
-- Performing Test SCOTCH_Num_8
-- Performing Test SCOTCH_Num_8 - Failed
-- A version of Pastix has been found but pastix_nompi.h does not exist in the include directory. Because Eigen tests require a version without MPI, we disable the Pastix backend.
-- Found SPQR: /usr/include  
-- Found unsuitable Qt version "5.15.6" from /usr/bin/qmake
-- Performing Test COMPILER_SUPPORT_FASTMATH
-- Performing Test COMPILER_SUPPORT_FASTMATH - Success
-- Found Boost: /usr/lib64/cmake/Boost-1.80.0/BoostConfig.cmake (found suitable version "1.80.0", minimum required is "1.53.0")  
-- Found Threads: TRUE  
-- Could NOT find GoogleHash (missing: GOOGLEHASH_INCLUDES GOOGLEHASH_COMPILE) 
-- Could NOT find Adolc (missing: ADOLC_INCLUDES ADOLC_LIBRARIES) 
-- Found MPFR: /usr/include (Required is at least version "1.0.0") 
-- Found GMP: /usr/include  
-- Could NOT find MPREAL (missing: MPREAL_INCLUDES MPREAL_VERSION_OK MPREAL_WORKS) (Required is at least version "1.0.0")
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
-- Found FFTW: /usr/include  
CUDA_TOOLKIT_ROOT_DIR not found or specified
-- Could NOT find CUDA (missing: CUDA_TOOLKIT_ROOT_DIR CUDA_NVCC_EXECUTABLE CUDA_INCLUDE_DIRS CUDA_CUDART_LIBRARY) (Required is at least version "7.0")
-- Found unsuitable Qt version "5.15.6" from /usr/bin/qmake
-- Qt4 not found, so disabling the mandelbrot and opengl demos
-- Could NOT find SuperLU (missing: SUPERLU_INCLUDES SUPERLU_LIBRARIES SUPERLU_VERSION_OK) (Required is at least version "4.0")
-- A version of Pastix has been found but pastix_nompi.h does not exist in the include directory. Because Eigen tests require a version without MPI, we disable the Pastix backend.
-- ************************************************************
-- ***    Eigen's unit tests configuration summary          ***
-- ************************************************************
-- 
-- Build type:        Release
-- Build site:        linux
-- Build string:      linux-5.19.10-arch1-1-g++-12.2.0-sse2-64bit
-- Enabled backends:  CHOLMOD,  UMFPACK,  KLU,  SPQR,  Boost.Multiprecision,  fftw, 
-- Disabled backends: SuperLU,  PaStiX,  METIS,  Qt4 support,  GoogleHash,  Adolc,  MPFR C++,  OpenGL, 
-- Default order:     Column-major
-- Maximal matrix/vector size: 320
-- SSE2:              Using architecture defaults
-- SSE3:              Using architecture defaults
-- SSSE3:             Using architecture defaults
-- SSE4.1:            Using architecture defaults
-- SSE4.2:            Using architecture defaults
-- AVX:               Using architecture defaults
-- AVX2:              Using architecture defaults
-- FMA:               Using architecture defaults
-- AVX512:            Using architecture defaults
-- AVX512DQ:          Using architecture defaults
-- Altivec:           Using architecture defaults
-- VSX:               Using architecture defaults
-- MIPS MSA:          Using architecture defaults
-- ARM NEON:          Using architecture defaults
-- ARMv8 NEON:        Using architecture defaults
-- S390X ZVECTOR:     Using architecture defaults
-- C++11:             OFF
-- SYCL:              OFF
-- CUDA:              OFF
-- HIP:               OFF
-- 
CXX:               /usr/bin/c++
 CXX_VERSION:       c++ (GCC) 12.2.0
 CXX_FLAGS:          -std=c++03 -pedantic -Wall -Wextra -Wundef -Wcast-align -Wchar-subscripts -Wnon-virtual-dtor -Wunused-local-typedefs -Wpointer-arith -Wwrite-strings -Wformat-security -Wlogical-op -Wenum-conversion -Wc++11-extensions -Wdouble-promotion -Wshadow -Wno-psabi -Wno-variadic-macros -Wno-long-long -fno-check-new -fno-common -fstrict-aliasing -ansi
 Sparse lib flags:   ;/usr/lib/libcholmod.so;/usr/lib/libamd.so;/usr/lib/libcolamd.so;/usr/lib/libcamd.so;/usr/lib/libccolamd.so;/usr/lib/libmetis.so;eigen_blas;eigen_lapack;/usr/lib/libumfpack.so;/usr/lib/libcolamd.so;/usr/lib/libamd.so;/usr/lib/libcholmod.so;eigen_blas;/usr/lib/libklu.so;/usr/lib/libcolamd.so;/usr/lib/libamd.so;/usr/lib/libbtf.so;eigen_blas;/usr/lib/libspqr.so;/usr/lib/libcholmod.so;/usr/lib/libcholmod.so;/usr/lib/libamd.so;/usr/lib/libcolamd.so;/usr/lib/libcamd.so;/usr/lib/libccolamd.so;/usr/lib/libmetis.so;eigen_lapack;eigen_blas

-- ************************************************************
-- 
-- Configured Eigen 3.4.0
-- 
-- Available targets (use: make TARGET):
-- ---------+--------------------------------------------------------------
-- Target   |   Description
-- ---------+--------------------------------------------------------------
-- install  | Install Eigen. Headers will be installed to:
--          |     <CMAKE_INSTALL_PREFIX>/<INCLUDE_INSTALL_DIR>
--          |   Using the following values:
--          |     CMAKE_INSTALL_PREFIX: /usr/local
--          |     INCLUDE_INSTALL_DIR:  include/eigen3
--          |   Change the install location of Eigen headers using:
--          |     cmake . -DCMAKE_INSTALL_PREFIX=yourprefix
--          |   Or:
--          |     cmake . -DINCLUDE_INSTALL_DIR=yourdir
-- doc      | Generate the API documentation, requires Doxygen & LaTeX
-- check    | Build and run the unit-tests. Read this page:
--          |   http://eigen.tuxfamily.org/index.php?title=Tests
-- blas     | Build BLAS library (not the same thing as Eigen)
-- uninstall| Remove files installed by the install target
-- ---------+--------------------------------------------------------------
-- 
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- LTO enabled
-- Configuring done
-- Generating done
-- Build files have been written to: /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build
Building CXX object _deps/cnpy-build/CMakeFiles/cnpy.dir/cnpy.cpp.o
Linking CXX shared library libcnpy.so
Built target cnpy
Building CXX object CMakeFiles/wavernn.dir/wavernn.cpp.o
Building CXX object CMakeFiles/wavernn.dir/net_impl.cpp.o
Linking CXX static library libwavernn.a
Built target wavernn
Building CXX object CMakeFiles/vocoder.dir/vocoder.cpp.o
Linking CXX executable vocoder
Built target vocoder
Building CXX object CMakeFiles/WaveRNNVocoder.dir/WaveRNNVocoder.cpp.o
In file included from /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/cast.h:16,
                 from /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/attr.h:13,
                 from /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h:43,
                 from /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/runtimeracer_version/src/WaveRNNVocoder.cpp:2:
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/detail/internals.h:82:33: warning: ‘int PyThread_create_key()’ is deprecated [-Wdeprecated-declarations]
   82 |     decltype(PyThread_create_key()) tstate = 0; // Usually an int but a long on Cygwin64 with Python 3.x
      |              ~~~~~~~~~~~~~~~~~~~^~
In file included from /usr/include/python3.10/Python.h:122,
                 from /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/detail/common.h:111,
                 from /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pytypes.h:12,
                 from /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/cast.h:13:
/usr/include/python3.10/pythread.h:108:36: note: declared here
  108 | Py_DEPRECATED(3.7) PyAPI_FUNC(int) PyThread_create_key(void);
      |                                    ^~~~~~~~~~~~~~~~~~~
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/detail/internals.h:82:33: warning: ‘int PyThread_create_key()’ is deprecated [-Wdeprecated-declarations]
   82 |     decltype(PyThread_create_key()) tstate = 0; // Usually an int but a long on Cygwin64 with Python 3.x
      |              ~~~~~~~~~~~~~~~~~~~^~
/usr/include/python3.10/pythread.h:108:36: note: declared here
  108 | Py_DEPRECATED(3.7) PyAPI_FUNC(int) PyThread_create_key(void);
      |                                    ^~~~~~~~~~~~~~~~~~~
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/detail/internals.h: In function ‘pybind11::detail::internals& pybind11::detail::get_internals()’:
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/detail/internals.h:167:27: warning: ‘void PyEval_InitThreads()’ is deprecated [-Wdeprecated-declarations]
  167 |         PyEval_InitThreads();
      |         ~~~~~~~~~~~~~~~~~~^~
In file included from /usr/include/python3.10/Python.h:130:
/usr/include/python3.10/ceval.h:122:37: note: declared here
  122 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
      |                                     ^~~~~~~~~~~~~~~~~~
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/detail/internals.h:169:52: warning: ‘int PyThread_create_key()’ is deprecated [-Wdeprecated-declarations]
  169 |         internals_ptr->tstate = PyThread_create_key();
      |                                 ~~~~~~~~~~~~~~~~~~~^~
/usr/include/python3.10/pythread.h:108:36: note: declared here
  108 | Py_DEPRECATED(3.7) PyAPI_FUNC(int) PyThread_create_key(void);
      |                                    ^~~~~~~~~~~~~~~~~~~
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/detail/internals.h:170:31: warning: ‘int PyThread_set_key_value(int, void*)’ is deprecated [-Wdeprecated-declarations]
  170 |         PyThread_set_key_value(internals_ptr->tstate, tstate);
      |         ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.10/pythread.h:110:36: note: declared here
  110 | Py_DEPRECATED(3.7) PyAPI_FUNC(int) PyThread_set_key_value(int key,
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h: In constructor ‘pybind11::gil_scoped_acquire::gil_scoped_acquire()’:
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h:1750:58: warning: ‘void* PyThread_get_key_value(int)’ is deprecated [-Wdeprecated-declarations]
 1750 |         tstate = (PyThreadState *) PyThread_get_key_value(internals.tstate);
      |                                    ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/usr/include/python3.10/pythread.h:112:39: note: declared here
  112 | Py_DEPRECATED(3.7) PyAPI_FUNC(void *) PyThread_get_key_value(int key);
      |                                       ^~~~~~~~~~~~~~~~~~~~~~
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h:1762:35: warning: ‘int PyThread_set_key_value(int, void*)’ is deprecated [-Wdeprecated-declarations]
 1762 |             PyThread_set_key_value(internals.tstate, tstate);
      |             ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.10/pythread.h:110:36: note: declared here
  110 | Py_DEPRECATED(3.7) PyAPI_FUNC(int) PyThread_set_key_value(int key,
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h: In member function ‘void pybind11::gil_scoped_acquire::dec_ref()’:
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h:1801:38: warning: ‘void PyThread_delete_key_value(int)’ is deprecated [-Wdeprecated-declarations]
 1801 |             PyThread_delete_key_value(detail::get_internals().tstate);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/python3.10/pythread.h:113:37: note: declared here
  113 | Py_DEPRECATED(3.7) PyAPI_FUNC(void) PyThread_delete_key_value(int key);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h: In constructor ‘pybind11::gil_scoped_release::gil_scoped_release(bool)’:
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h:1829:39: warning: ‘int PyThread_set_key_value(int, void*)’ is deprecated [-Wdeprecated-declarations]
 1829 |                 PyThread_set_key_value(key, nullptr);
      |                 ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
/usr/include/python3.10/pythread.h:110:36: note: declared here
  110 | Py_DEPRECATED(3.7) PyAPI_FUNC(int) PyThread_set_key_value(int key,
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h: In destructor ‘pybind11::gil_scoped_release::~gil_scoped_release()’:
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/build/_deps/pybind11-src/include/pybind11/pybind11.h:1842:35: warning: ‘int PyThread_set_key_value(int, void*)’ is deprecated [-Wdeprecated-declarations]
 1842 |             PyThread_set_key_value(key, tstate);
      |             ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
/usr/include/python3.10/pythread.h:110:36: note: declared here
  110 | Py_DEPRECATED(3.7) PyAPI_FUNC(int) PyThread_set_key_value(int key,
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
In file included from /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/runtimeracer_version/src/net_impl.h:6,
                 from /home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/runtimeracer_version/src/WaveRNNVocoder.cpp:8:
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/runtimeracer_version/src/wavernn.h: In member function ‘void CompMatrix::prepData(std::vector<float>&, std::vector<unsigned char>&)’:
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/runtimeracer_version/src/wavernn.h:39:43: error: ‘aligned_alloc’ is not a member of ‘std’; did you mean ‘aligned_union’?
   39 |         weight = static_cast<float*>(std::aligned_alloc(32, sizeof(float)*wght.size()));
      |                                           ^~~~~~~~~~~~~
      |                                           aligned_union
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/runtimeracer_version/src/wavernn.h:40:41: error: ‘aligned_alloc’ is not a member of ‘std’; did you mean ‘aligned_union’?
   40 |         rowIdx = static_cast<int*>(std::aligned_alloc(32, sizeof(int)*nGroups));
      |                                         ^~~~~~~~~~~~~
      |                                         aligned_union
/home/neo/experiments/RT2/Real-Time-Voice-Cloning/vocoder/libwavernn/runtimeracer_version/src/wavernn.h:41:44: error: ‘aligned_alloc’ is not a member of ‘std’; did you mean ‘aligned_union’?
   41 |         colIdx = static_cast<int8_t*>(std::aligned_alloc(32, sizeof(int8_t)*nGroups));
      |                                            ^~~~~~~~~~~~~
      |                                            aligned_union
make[2]: *** [CMakeFiles/WaveRNNVocoder.dir/build.make:76: CMakeFiles/WaveRNNVocoder.dir/WaveRNNVocoder.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3233: CMakeFiles/WaveRNNVocoder.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
cp: cannot stat 'build/WaveRNNVocoder*.so': No such file or directory
Defaulting to user installation because normal site-packages is not writeable
Processing ./WaveRNNVocoder
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: WaveRNNVocoder
  Building wheel for WaveRNNVocoder (setup.py) ... done
  Created wheel for WaveRNNVocoder: filename=WaveRNNVocoder-0.1-py3-none-any.whl size=1057 sha256=b6d623cd3066df6600b6c8464d0eaa2b8ba68ff6110334921bead04a3e7701b7
  Stored in directory: /tmp/pip-ephem-wheel-cache-9fz2z699/wheels/da/c0/7c/f26e17ae579b8c98286526d532caff4234695691ae26594902
Successfully built WaveRNNVocoder
Installing collected packages: WaveRNNVocoder
Successfully installed WaveRNNVocoder-0.1

```

trholding commented:
Fix:

For my case (x86_64 linux, gcc 12.2.0)

In wavernn.h changed std::aligned_alloc to alligned_alloc

```
#ifdef __linux__
        weight = static_cast<float*>(aligned_alloc(32, sizeof(float)*wght.size()));
        rowIdx = static_cast<int*>(aligned_alloc(32, sizeof(int)*nGroups));
        colIdx = static_cast<int8_t*>(aligned_alloc(32, sizeof(int8_t)*nGroups));
#elif _WIN32
```

trholding commented:
Okay, got so far:

I guess the model files from CorentinJ/ are not valid for this one...

@RuntimeRacer would be cool if you could share the pre-trained models :)

```
python demo_cli.py 
<frozen importlib._bootstrap>:241: FutureWarning: pybind11-bound class 'WaveRNNVocoder.Matrix' is using an old-style placement-new '__init__' which has been deprecated. See the upgrade guide in pybind11's docs. This message is only visible when compiled in debug mode.
Arguments:
    enc_model_fpath:   encoder/saved_models/pretrained.pt
    syn_model_fpath:   synthesizer/saved_models/pretrained/pretrained.pt
    voc_model_fpath:   vocoder/saved_models/pretrained/pretrained.pt
    cpu:               False
    no_sound:          False
    seed:              None
    no_mp3_support:    False

/home/linux/.local/lib/python3.10/site-packages/librosa/util/decorators.py:88: UserWarning: PySoundFile failed. Trying audioread instead.
  return f(*args, **kwargs)
Running a test of your configuration...

Using CPU for inference.

Preparing the encoder, the synthesizer and the vocoder...
Traceback (most recent call last):
  File "/home/neo/experiments/RT2/Real-Time-Voice-Cloning/demo_cli.py", line 80, in <module>
    encoder.load_model(args.enc_model_fpath)
  File "/home/neo/experiments/RT2/Real-Time-Voice-Cloning/encoder/inference.py", line 35, in load_model
    _model.load_state_dict(checkpoint["model_state"])
  File "/home/linux/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1604, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for SpeakerEncoder:
        size mismatch for lstm.weight_ih_l0: copying a param with shape torch.Size([1024, 40]) from checkpoint, the shape in current model is torch.Size([3072, 40]).
        size mismatch for lstm.weight_hh_l0: copying a param with shape torch.Size([1024, 256]) from checkpoint, the shape in current model is torch.Size([3072, 768]).
        size mismatch for lstm.bias_ih_l0: copying a param with shape torch.Size([1024]) from checkpoint, the shape in current model is torch.Size([3072]).
        size mismatch for lstm.bias_hh_l0: copying a param with shape torch.Size([1024]) from checkpoint, the shape in current model is torch.Size([3072]).
        size mismatch for lstm.weight_ih_l1: copying a param with shape torch.Size([1024, 256]) from checkpoint, the shape in current model is torch.Size([3072, 768]).
        size mismatch for lstm.weight_hh_l1: copying a param with shape torch.Size([1024, 256]) from checkpoint, the shape in current model is torch.Size([3072, 768]).
        size mismatch for lstm.bias_ih_l1: copying a param with shape torch.Size([1024]) from checkpoint, the shape in current model is torch.Size([3072]).
        size mismatch for lstm.bias_hh_l1: copying a param with shape torch.Size([1024]) from checkpoint, the shape in current model is torch.Size([3072]).
        size mismatch for lstm.weight_ih_l2: copying a param with shape torch.Size([1024, 256]) from checkpoint, the shape in current model is torch.Size([3072, 768]).
        size mismatch for lstm.weight_hh_l2: copying a param with shape torch.Size([1024, 256]) from checkpoint, the shape in current model is torch.Size([3072, 768]).
        size mismatch for lstm.bias_ih_l2: copying a param with shape torch.Size([1024]) from checkpoint, the shape in current model is torch.Size([3072]).
        size mismatch for lstm.bias_hh_l2: copying a param with shape torch.Size([1024]) from checkpoint, the shape in current model is torch.Size([3072]).
        size mismatch for linear.weight: copying a param with shape torch.Size([256, 256]) from checkpoint, the shape in current model is torch.Size([768, 768]).
        size mismatch for linear.bias: copying a param with shape torch.Size([256]) from checkpoint, the shape in current model is torch.Size([768]).
```

RuntimeRacer commented:
Hello, hello!

@trholding hi I wasn't aware that my fork was on someone else's radar already; I wanted to pull everything straight at some point in the near future when I fixed a couple of things, added proper documentation etc. and merged back the current master from CorentinJ so it's compatible again.

So first of all, I never used the CLI when testing and always used the toolbox; so I would assume some things will not work when using the CLI. Especially in case you want to use it with a libwavernn implementation, you'll need to switch the inference processor from pytorch to `WaveRNNVocoder`, after building it and installing it as a pip module (this is what my `build.sh` actually does).

However I am building with conda; adding a switch to build with local python in case someone is not using conda is one of the things which I wanted to pull straight first, for example. (I'd also recommend using conda or venv badly; so you can easily switch between environments and library versions in these environments. Conda has the advantage that it's not bound to only python also).

Having that said, if you want to use libwavernn; you'll need to:

1. Train a vocoder model with the adapted pytorch models in this repo + using pruning on it (otherwise no speed gain during inference). I pulled in the libwavernn code from another repo; it requires changing the input aux channel format to work. I did that for all models in this repo.
2. Build the pip extension and install it using `build.sh`
3. After training + finished pruning, convert it to binary using the conversion helper
4. Either use toolbox and enable the switch for libwavernn, or somehow integrate it in the cli variant.

Regarding pre-trained models; I can share those I have later maybe; but not sure if they will be compatible with mel spectograms from CorentinJs pretrained synthesizer. Maybe they are, but I might have changed some preprocessing and output parameters when working with it.

Also I wasn't too happy with either performance or quality when pruned, so I am now working on integrating Multiband MelGAN in a seperate branch, but it will take some time until that is done.

Hope this was at least a bit helpful. In case I forget sharing the models, feel free to ping me a bit later again.

Cheers, RR

RuntimeRacer commented:
> Fix:
> 
> For my case (x86_64 linux, gcc 12.2.0)
> 
> In wavernn.h changed std::aligned_alloc to alligned_alloc
> 
> ```
> #ifdef __linux__
>         weight = static_cast<float*>(aligned_alloc(32, sizeof(float)*wght.size()));
>         rowIdx = static_cast<int*>(aligned_alloc(32, sizeof(int)*nGroups));
>         colIdx = static_cast<int8_t*>(aligned_alloc(32, sizeof(int8_t)*nGroups));
> #elif _WIN32
> ```

This seems a bit weird for me. Maybe you got overlapping namespaces or sth? Because `aligned_alloc` is always part of the `std` C++ library; it should make no difference whether `std::` is explicitly there.

RuntimeRacer commented:
Btw for better visibility, maybe open followup messages for this conversation rather in the issues tab of the repo instead of commenting to an already closed PR.

RuntimeRacer commented:
Ok my bad; I did not enable issues for this repo. enabled it now.
